src/main/java/org/terasology/scenario/internal/systems/ActionEventSystem.java

```diff
@@ -158,4 +177,129 @@ public void onEventTriggerEvent(EventTriggerEvent event, EntityRef entity, Scena
+        variables.get("block").send(evaluateBlockEvent);
+        BlockFamily blockFamily = evaluateBlockEvent.getResult();
+
+        EvaluateIntEvent intEvaluateEvent = new EvaluateIntEvent(event.informationEntity);
+        variables.get("amount").send(intEvaluateEvent);
+        int amount = intEvaluateEvent.getResult();
+
+        ScenarioValuePlayerComponent.PlayerType player = variables.get("player").getComponent(ScenarioValuePlayerComponent.class).type;
+
+        if (player == ScenarioValuePlayerComponent.PlayerType.TRIGGERING_PLAYER) {
+            EntityRef playerEnt = event.informationEntity.getComponent(InfoTriggeringEntityComponent.class).entity;
+
+            List<EntityRef> items = playerEnt.getComponent(InventoryComponent.class).itemSlots;
+
+            for (EntityRef e : items) {
+                if (e.hasComponent(BlockItemComponent.class)) {
+                    if (e.getComponent(BlockItemComponent.class).blockFamily.equals(blockFamily)) {
+                        ItemComponent itemComponent = e.getComponent(ItemComponent.class);
+                        if (itemComponent.stackCount > amount) {
+                            itemComponent.stackCount-= amount;
+                            e.saveComponent(itemComponent);
+                            break;
+                        }
+                        else if (itemComponent.stackCount == amount) {
+                            e.destroy();
+                            break;
+                        }
+                        else { //Same item, but stack isn't big enough
+                            amount -= itemComponent.stackCount;
+                            e.destroy();
+                        }
+                    }
+                }
+            }
+
+            playerEnt.saveComponent(playerEnt.getComponent(InventoryComponent.class));
+        }
+    }
+
+    @ReceiveEvent //Heal Player
+    public void onEventTriggerEvent(EventTriggerEvent event, EntityRef entity, ScenarioSecondaryHealComponent action) {
+        Map<String, EntityRef> variables = entity.getComponent(ScenarioArgumentContainerComponent.class).arguments;
+
+        EvaluateIntEvent intEvaluateEvent = new EvaluateIntEvent(event.informationEntity);
+        variables.get("amount").send(intEvaluateEvent);
+        int amount = intEvaluateEvent.getResult();
+
+        ScenarioValuePlayerComponent.PlayerType player = variables.get("player").getComponent(ScenarioValuePlayerComponent.class).type;
+        if (player == ScenarioValuePlayerComponent.PlayerType.TRIGGERING_PLAYER) {
+            EntityRef playerEnt = event.informationEntity.getComponent(InfoTriggeringEntityComponent.class).entity;
+            playerEnt.send(new DoHealEvent(amount, playerEnt));
+        }
+    }
+
+    @ReceiveEvent //Damage Player
+    public void onEventTriggerEvent(EventTriggerEvent event, EntityRef entity, ScenarioSecondaryDamageComponent action) {

```

Comment:
I think damage you often more likely want do do in absolute values. Maybe you want to name the component so that it indicates that it is for percentage damage. Just setting health is something better don in percentage values